### PR TITLE
Go install

### PR DESCRIPTION
**Pull request by Tatskaari**

### CODE REVIEW

File: tools/please_go_install/main.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	fmt.Println(prepOutDir)
+	fmt.Println(compile)
+	fmt.Println(updateImportCfg)
+
+	if binary {
+		goToolLink(out)
+	}
+}
+
+func goToolCGOCompile(target string, binary bool, pkgDir string, srcs []string, cgoSrcs []string, pkg *ast.Package) {
```
Comment (Tatskaari) [inline tools/please_go_install/main.go:192]: This produces a script a bit like:
```
#!/bin/sh
set -e
mkdir -p pkg/linux_amd64/github.com/DataDog
ln importconfig third_party/go/src/github.com/DataDog/zstd
cd third_party/go/src/github.com/DataDog/zstd
/usr/bin/go tool cgo zstd_stream.go zstd.go errors.go
/usr/bin/go tool compile -pack -importcfg importconfig -o out.a _obj/*.go
/usr/bin/gcc -Wno-error -ldl -Wno-unused-parameter -c -I _obj -I . *.c
/usr/bin/gcc -Wno-error -ldl -Wno-unused-parameter -c -I _obj -I . _obj/_cgo_export.c _obj/*.cgo2.c
/usr/bin/go tool pack r out.a *.o 
cd - && mv third_party/go/src/github.com/DataDog/zstd/out.a pkg/linux_amd64/github.com/DataDog/zstd.a
echo "packagefile github.com/DataDog/zstd=pkg/linux_amd64/github.com/DataDog/zstd.a" >> importconfig
```

File: docs/styles.css [hunk 2]
```diff
@@ -701,7 +701,7 @@ footer > div {
 
 /* codelabs */
 .code-lab {
-    margin: 3em 1em 1em;
+    margin: 3em 1em 15px -15px;
```
Comment (peterebden) [inline docs/styles.css:704]: this will definitely improve `go install`

**Review by peterebden — APPROVED** [review]
Nice work - I feel there could be some ongoing edge cases but conceptually seems like it's probably the right direction.

File: docs/styles.css (hunk 2 above)
Comment (Tatskaari) [reply docs/styles.css:704]: Indeed this is the kingpin holding the whole thing together!

**Pull request merged**